code/modules/hydroponics/gene_modder.dm

```diff
@@ -107,258 +106,135 @@
+			)
+			data["reagent_genes"] += list(seed_info)
+			data["has_reagent"] = TRUE
 
-	dat += "<br></div>"
+		for(var/datum/plant_gene/trait/t_gene in trait_genes)
+			var/list/seed_info = list(
+				"name" = t_gene.get_name(),
+				"id" = t_gene.UID()
+			)
+			data["trait_genes"] += list(seed_info)
+			data["has_trait"] = TRUE
 
-	if(seed)
-		var/can_insert = disk && disk.gene && disk.gene.can_add(seed)
-		var/can_extract = disk && !disk.read_only
-
-		dat += "<div class='line'><h3>Core Genes</h3></div><div class='statusDisplay'><table>"
-		for(var/a in core_genes)
-			var/datum/plant_gene/G = a
-			if(!G)
-				continue
-			dat += "<tr><td width='260px'>[G.get_name()]</td><td>"
-			if(can_extract)
-				dat += "<a href='?src=[UID()];gene=[G.UID()];op=extract'>Extract</a>"
-			if(can_insert && istype(disk.gene, G.type))
-				dat += "<a href='?src=[UID()];gene=[G.UID()];op=replace'>Replace</a>"
-			dat += "</td></tr>"
-		dat += "</table></div>"
-
-		if(seed.yield != -1)
-			dat += "<div class='line'><h3>Content Genes</h3></div><div class='statusDisplay'>"
-			if(reagent_genes.len)
-				dat += "<table>"
-				for(var/a in reagent_genes)
-					var/datum/plant_gene/G = a
-					dat += "<tr><td width='260px'>[G.get_name()]</td><td>"
-					if(can_extract)
-						dat += "<a href='?src=[UID()];gene=[G.UID()];op=extract'>Extract</a>"
-					dat += "<a href='?src=[UID()];gene=[G.UID()];op=remove'>Remove</a>"
-					dat += "</td></tr>"
-				dat += "</table>"
-			else
-				dat += "No content-related genes detected in sample.<br>"
-			dat += "</div>"
-			if(can_insert && istype(disk.gene, /datum/plant_gene/reagent))
-				dat += "<a href='?src=[UID()];op=insert'>Insert: [disk.gene.get_name()]</a>"
-
-			dat += "<div class='line'><h3>Trait Genes</h3></div><div class='statusDisplay'>"
-			if(trait_genes.len)
-				dat += "<table>"
-				for(var/a in trait_genes)
-					var/datum/plant_gene/G = a
-					dat += "<tr><td width='260px'>[G.get_name()]</td><td>"
-					if(can_extract)
-						dat += "<a href='?src=[UID()];gene=[G.UID()];op=extract'>Extract</a>"
-					dat += "<a href='?src=[UID()];gene=[G.UID()];op=remove'>Remove</a>"
-					dat += "</td></tr>"
-				dat += "</table>"
-			else
-				dat += "No trait-related genes detected in sample.<br>"
-			if(can_insert && istype(disk.gene, /datum/plant_gene/trait))
-				dat += "<a href='?src=[UID()];op=insert'>Insert: [disk.gene.get_name()]</a>"
-			dat += "</div>"
-
-		dat += "<div class='line'><h3>Variant</h3></div><div class='statusDisplay'><table>"
-		dat += "<tr><td width='260px'>[seed.variant ? seed.variant : "None"]</td>"
-		dat += "<td><a href='?src=[UID()];set_v=1'>Edit</a></td>"
-		if(seed.variant)
-			dat += "<td><a href='?src=[UID()];del_v=1'>Remove</a></td>"
-		dat += "</tr></table></div>"
-	else
-		dat += "<br>No sample found.<br><span class='highlight'>Please, insert a plant sample to use this device.</span>"
-	popup.set_content(dat)
-	popup.open()
+	data["disk"] = list()
+
+	if(disk)
+		var/name = "Empty Disk"
```

Comment:
you should pick a different var name than `name` since it conflicts with the base `name` variable on all atoms